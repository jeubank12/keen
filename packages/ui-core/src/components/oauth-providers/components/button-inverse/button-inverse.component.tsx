import styled from 'styled-components';
import { colors } from '@keen.io/colors';
import { transparentize } from 'polished';

export const ButtonInverse = styled.div`
  display: flex;
  flex-shrink: 0;
  align-items: center;
  justify-content: center;

  border-radius: 5rem;
  border: 1px solid ${colors.white['300']};
  cursor: pointer;

  font-family: 'Lato Regular', sans-serif;

  box-shadow: 0 1px 4px 0 ${transparentize(0.85, colors.black['500'])};

  padding: 12px 16px;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  background: transparent;
  color: ${colors.blue['500']};

  &:hover {
    background: #f2f2f2;
    box-shadow: none;
  }

  transition: background 0.2s linear;
`;
