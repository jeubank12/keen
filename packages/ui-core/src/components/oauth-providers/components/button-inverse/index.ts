import { ButtonInverse } from './button-inverse.component';

export default ButtonInverse;
