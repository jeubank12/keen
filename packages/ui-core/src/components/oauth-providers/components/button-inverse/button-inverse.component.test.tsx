import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ButtonInverse } from './button-inverse.component';

const render = (children: React.ReactNode) => {
  const wrapper = rtlRender(<ButtonInverse>{children}</ButtonInverse>);

  return {
    wrapper,
  };
};

describe('<ButtonInverse />', () => {
  const children = <span>Sign up</span>;
  test('should render button correctly', () => {
    const {
      wrapper: { container },
    } = render(children);

    expect(container).toMatchSnapshot();
  });
});
