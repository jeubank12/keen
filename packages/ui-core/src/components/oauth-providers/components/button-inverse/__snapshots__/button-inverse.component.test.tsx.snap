// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonInverse /> should render button correctly 1`] = `
<div>
  <div
    class="sc-AxjAm kxAyDr"
  >
    <span>
      Sign up
    </span>
  </div>
</div>
`;
